frontend/frontend/protocols/ssh/_proxy_handler.py

```diff
@@ -0,0 +1,324 @@
+"""This file contains the ProxyHandler class which can be used to
+proxy data between the backend and log commands recieved from the attacker and
+all the data recieved from the backend during the sesison.
+"""
+import logging
+import socket
+import threading
+from time import sleep
+
+import paramiko
+from paramiko import SSHException
+from paramiko.ssh_exception import AuthenticationException
+from paramiko.transport import Transport
+from paramiko.channel import Channel
+from paramiko.common import (OPEN_FAILED_ADMINISTRATIVELY_PROHIBITED, OPEN_SUCCEEDED)
+
+from frontend.honeylogger import SSHSession
+from ._command_parser import CommandParser
+
+
+logger = logging.getLogger(__name__)
+
+
+class ProxyHandler:
```

Comment:
This class proxies all data received from an attacker to another SSH connection. Most of the methods here are called from ssh_server when data there is received and it is sent to this class to proxy further